### PR TITLE
Inject CatalogName in jdbc connectors

### PR DESCRIPTION
**Pull request by kokosing**

### CODE REVIEW

File: presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/JdbcConnectorFactory.java [hunk 1]
```diff
@@ -40,9 +40,14 @@
 
     public JdbcConnectorFactory(String name, Module module)
     {
-        this(name, JdbcModuleProvider.withCredentialProvider(module));
```
Comment (sopel39) [inline presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/JdbcConnectorFactory.java:43]: should we hide constructors (make them protected) of `JdbcConnectorFactory` and use static method instead? It's not obvious why `module` should be wrapped in `withCredentialProvider` and this commit changes mechanics of `JdbcConnectorFactory`

Comment (kokosing) [reply presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/JdbcConnectorFactory.java:43]: Hiding constructors is quite invasive change.

You should use `withCredentialProvider` when connector expects user should pass `user` and `password` in config properties. 

It is a typical jdbc connector behaviour. Notice that typically connector extend `io.prestosql.plugin.jdbc.JdbcPlugin` and from that point of view nothing changes.

File: presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/JdbcConnectorFactory.java [hunk 2]
```diff
@@ -83,14 +88,14 @@ public Connector create(String catalogName, Map<String, String> requiredConfig,
         return injector.getInstance(JdbcConnector.class);
     }
 
+    @Deprecated
```
Comment (findepi) [inline presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/JdbcConnectorFactory.java:91]: add `/** @deprecated`  and tell what to use instead.

also the commit message ("Deprecate JdbcModuleProvider") could indicate  a hint as well

File: presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/JdbcConnectorFactory.java [hunk 3]
```diff
@@ -40,9 +40,14 @@
 
     public JdbcConnectorFactory(String name, Module module)
     {
-        this(name, JdbcModuleProvider.withCredentialProvider(module));
+        this(name, catalogName -> module);
     }
 
+    /**
+     * Prefer {@link JdbcConnectorFactory#JdbcConnectorFactory(String, Module)}.
```
Comment (findepi) [inline presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/JdbcConnectorFactory.java:47]: make it into `@deprecated`

File: presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/JdbcConnectorFactory.java [hunk 4]
```diff
@@ -40,9 +40,14 @@
...
+     * Notice that {@link io.prestosql.plugin.base.CatalogName} is available in guice context.
+     */
+    @Deprecated
```
Comment (findepi) [inline presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/JdbcConnectorFactory.java:50]: this is still _the constructor_ (with the other one delegating here).

Will we have this deprecated staying here & being used for ever?

File: presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/JdbcConnectorFactory.java [hunk 5]
```diff
@@ -83,14 +88,14 @@ public Connector create(String catalogName, Map<String, String> requiredConfig,
...
     {
         Module getModule(String catalogName);
+    }
 
-        static JdbcModuleProvider withCredentialProvider(Module module)
-        {
-            requireNonNull(module, "module is null");
-            return catalogName -> combine(new CredentialProviderModule(), module);
-        }
+    public static Module withCredentialProvider(Module module)
```
Comment (findepi) [inline presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/JdbcConnectorFactory.java:97]: You're not maintaining compatibility (the FQN name of the method changed), so you can as well remove this method altogether (inline). I think we should rather pull `CredentialProviderModule` explicitly.

File: presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/JdbcDiagnosticModule.java [hunk 6]
```diff
@@ -28,35 +30,28 @@
...
-        this.catalogName = requireNonNull(catalogName, "catalogName is null");
-    }
-
     @Override
     public void configure(Binder binder)
     {
         binder.install(new MBeanServerModule());
         binder.install(new MBeanModule());
 
+        Provider<CatalogName> catalogName = binder.getProvider(CatalogName.class);
```
Comment (findepi) [inline presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/JdbcDiagnosticModule.java:44]: Squash with `Inject CatalogName in JDBC connectors`?

the `Inject CatalogName in JDBC connectors` commit looks so void on its own

**Review by findepi — APPROVED** [review]

File: presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/JdbcConnectorFactory.java (hunk 4 above)
Comment (kokosing) [reply presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/JdbcConnectorFactory.java:50]: Once we start to use the other constructor widely, we could remove this one and then we could refactor this class and remove `JdbcModuleProvider`.

**Review by kokosing — COMMENTED** [review]
AC

**Review by sopel39 — APPROVED** [review]

**Pull request merged**